scripts/TokenBank/minHash.cpp

```diff
@@ -0,0 +1,129 @@
+  // They're not really 'magic', they just happen to work well.
+
+  const uint32_t m = 0x5bd1e995;
+  const int r = 24;
+
+  // Initialize the hash to a 'random' value
+
+  uint32_t h = seed ^ len;
+
+  // Mix 4 bytes at a time into the hash
+
+  const unsigned char * data = (const unsigned char *)key;
+
+  while(len >= 4)
+  {
+    uint32_t k = *(uint32_t*)data;
+
+    k *= m;
+    k ^= k >> r;
+    k *= m;
+
+    h *= m;
+    h ^= k;
+
+    data += 4;
+    len -= 4;
+  }
+
+  // Handle the last few bytes of the input array
+
+  switch(len)
+  {
+  case 3: h ^= data[2] << 16;
+  case 2: h ^= data[1] << 8;
+  case 1: h ^= data[0];
+      h *= m;
+  };
+
+  // Do a few final mixes of the hash to ensure the last few
+  // bytes are well-incorporated.
+
+  h ^= h >> 13;
+  h *= m;
+  h ^= h >> 15;
+
+  return h;
+} 
+
+void minHash (const char* const iToken[], uint32_t iTokenLen[], uint32_t iTokenNum,
+    char* const oToken, uint32_t& oTokenLen) {
+  //minHash take three input (int str, int shingleLen, int b, int r)
+  assert(iTokenNum == 4);
+  uint32_t shingleLen = atoi(iToken[1]);
+  int b = atoi(iToken[2]);
+  int r = atoi(iToken[3]);
+  //only handle the case for the number of hash functions is less or equal to 100
+  assert(b*r <= 100);
+  uint32_t k = std::min(iTokenLen[0], shingleLen);
+  char tmp[k];
+  std::vector<uint32_t> signatures;
+  std::cout << "start signatures\n";
+  for (int i = 0; i < iTokenLen[0] - k + 1; ++i) {
+    signatures.push_back(MurmurHash2(&iToken[0][i], k, 0));
+    std::cout << "token: " << std::string(&iToken[0][i], k) << ", MurmurHash2: " <<  MurmurHash2(&iToken[0][i], k, 0) << std::endl;
+  }
+  std::cout << "end signatures\n";
+  //x = (a*x + b) % p
+  uint64_t p = 4294967311; //the prime number bigger than 2^32 - 1
+  uint32_t c1[101] = {1, 482067061, 1133999723, 2977257653, 2666089543, 3098200677, 3985189319, 1857983931, 3801236429, 522456919, 4057595205, 4176190031, 1652234975, 2294716503, 1644020081, 3407377875, 3749518465, 4244672803, 3053397733, 3273255487, 598272097, 3989043777, 1414109747, 697129027, 67285677, 98002333, 158583451, 1424122447, 2159224677, 3478101309, 277468927, 1902928727, 2459288935, 3941065327, 1244061689, 1898521317, 4205778491, 1987240989, 3446018461, 2407533397, 3151958893, 1553147067, 208156801, 2362352445, 2458343227, 4134443, 36216853, 932983869, 2800766507, 252990279, 2994662963, 2760285623, 4510445, 1458512423, 3500568231, 689831701, 887836659, 315834449, 2394075311, 1360826347, 439713319, 633358329, 749540625, 444867375, 531150885, 2871421439, 2347294453, 3975247983, 3255073387, 3561466319, 2616895667, 742825395, 3300710079, 1231551531, 3576325163, 3229203393, 2662941725, 3495109109, 2202779339, 2997513035, 1952088617, 2177967115, 1685362661, 2160536397, 2628206479, 1678152567, 775989269, 2114809421, 3882162141, 3267509575, 3869378355, 283353181, 306744579, 2793152333, 1454134621, 3021652657, 1664069155, 1711688171, 1264486497, 359065375, 1616608617};
```

Comment:
eliminate magic number: 101.

Define `const uint64_t HASH_FUNC_NUM=101;` at the beginning